pulsar-client-cpp/lib/MultiTopicsConsumerImpl.cc

```diff
@@ -711,3 +735,82 @@ uint64_t MultiTopicsConsumerImpl::getNumberOfConnectedConsumer() {
     });
     return numberOfConnectedConsumer;
 }
+void MultiTopicsConsumerImpl::runPartitionUpdateTask() {
+    partitionsUpdateTimer_->expires_from_now(partitionsUpdateInterval_);
+    auto self = shared_from_this();
+    partitionsUpdateTimer_->async_wait([self](const boost::system::error_code& ec) {
+        // If two requests call runPartitionUpdateTask at the same time, the timer will fail, and it
+        // cannot continue at this time, and the request needs to be ignored.
+        if (!ec) {
+            self->topicPartitionUpdate();
+        }
+    });
+}
+void MultiTopicsConsumerImpl::topicPartitionUpdate() {
+    using namespace std::placeholders;
+    for (const auto& item : topicsPartitions_) {
```

Comment:
`topicsPartitions_` should be protected by a mutex. I think you'd better to copy it into a local variable to avoid race condition.

```c++
Lock lock(mutex_);
auto topicsPartitions = topicsPartitions_;
lock.unlock();
for (const auto& item : topicsPartitions) {
```

Reply:
Agree. I changed it, thanks.